adaptors/library/jobs/DHIS2-DataValues-API.js

```diff
@@ -1,20 +1,33 @@
 // ----
 // Add data to data value sets in DHIS2 using a generic JSON message, submitted
-// by Taylor Downs @ OpenFn.
+// by Taylor Downs @ OpenFn. Co-authored by @mtuchi
 // ---
 
-dataValueSet(
-  fields(
-    field('dataSet', 'pBOMPrpg1QX'),
-    field('orgUnit', 'DiszpKrYNg8'),
-    field('period', '201401'),
-    field('completeData', dataValue('form.date')),
-    field('dataValues', function (state) {
-      return [
-        dataElement('qrur9Dvnyt5', dataValue('form.prop_a')(state)),
-        dataElement('oZg33kd9taw', dataValue('form.prop_b')(state)),
-        dataElement('msodh3rEMJa', dataValue('form.prop_c')(state)),
-      ];
-    })
-  )
-);
+fn(state => {
+  const { form } = state.data;
+  // Create a data value set mapping
+  state.dataValueSetMap = {
+    dataSet: 'pBOMPrpg1QX',
+    completeDate: form.date,
+    period: '201401',
+    orgUnit: 'DiszpKrYNg8',
+    dataValues: [
+      {
+        dataElement: 'f7n9E0hX8qk',
+        value: form.prop_a,
+      },
+      {
+        dataElement: 'Ix2HsbDMLea',
+        value: form.prop_b,
+      },
+      {
+        dataElement: 'eY5ehpbEsB7',
+        value: form.prop_c,
+      },
+    ],
+  };
+  return state;
+});
+
+// Create dataValueSets
+create('dataValueSets', state => state.dataValueSetMap);
```

Comment:
Wouldn't best practice be more like:
```
create('dataValueSets', {
  dataSet: 'pBOMPrpg1QX',
  completeDate: $.form.date,
  period: '201401',
  orgUnit: 'DiszpKrYNg8',
  dataValues: [
    {
      dataElement: 'f7n9E0hX8qk',
      value: $.form.prop_a,
    },
    {
      dataElement: 'Ix2HsbDMLea',
      value: $.form.prop_b,
    },
    {
      dataElement: 'eY5ehpbEsB7',
      value: $.form.prop_c,
    },
  ],
});
```

Reply:
That's clean and simple. I have updated the examples to use that pattern